src/mode/graphics.rs

```diff
@@ -113,75 +113,54 @@ where
+        match display_size {
+            DisplaySize::Display128x32 => Command::ComPinConfig(false, false).send(iface),
+            DisplaySize::Display128x64 => Command::ComPinConfig(true, false).send(iface),
+            DisplaySize::Display96x16 => Command::ComPinConfig(false, false).send(iface),
         }?;
 
-        Command::Contrast(0x8F).send(&mut self.iface)?;
-        Command::PreChargePeriod(0x1, 0xF).send(&mut self.iface)?;
-        Command::VcomhDeselect(VcomhLevel::Auto).send(&mut self.iface)?;
-        Command::AllOn(false).send(&mut self.iface)?;
-        Command::Invert(false).send(&mut self.iface)?;
-        Command::EnableScroll(false).send(&mut self.iface)?;
-        Command::DisplayOn(true).send(&mut self.iface)?;
-
-        Ok(())
-    }
-
-    fn configure_rotation(&mut self) -> Result<(), DI::Error> {
-        match self.display_rotation {
-            DisplayRotation::Rotate0 => {
-                Command::SegmentRemap(true).send(&mut self.iface)?;
-                Command::ReverseComDir(true).send(&mut self.iface)?;
-            }
-            DisplayRotation::Rotate90 => {
-                Command::SegmentRemap(false).send(&mut self.iface)?;
-                Command::ReverseComDir(true).send(&mut self.iface)?;
-            }
-            DisplayRotation::Rotate180 => {
-                Command::SegmentRemap(false).send(&mut self.iface)?;
-                Command::ReverseComDir(false).send(&mut self.iface)?;
-            }
-            DisplayRotation::Rotate270 => {
-                Command::SegmentRemap(true).send(&mut self.iface)?;
-                Command::ReverseComDir(false).send(&mut self.iface)?;
-            }
-        };
+        Command::Contrast(0x8F).send(iface)?;
+        Command::PreChargePeriod(0x1, 0xF).send(iface)?;
+        Command::VcomhDeselect(VcomhLevel::Auto).send(iface)?;
+        Command::AllOn(false).send(iface)?;
+        Command::Invert(false).send(iface)?;
+        Command::EnableScroll(false).send(iface)?;
+        Command::DisplayOn(true).send(iface)?;
 
         Ok(())
     }
 
     /// Get display dimensions, taking into account the current rotation of the display
     // TODO: Replace (u8, u8) with a dimensioney type for consistency
     pub fn get_dimensions(&self) -> (u8, u8) {
-        let (w, h) = self.display_size.dimensions();
+        let (w, h) = self.properties.get_size().dimensions();
+        let display_rotation = self.properties.get_rotation();
 
-        match self.display_rotation {
-            DisplayRotation::Rotate0 | DisplayRotation::Rotate180 => (w, h),
-            DisplayRotation::Rotate90 | DisplayRotation::Rotate270 => (h, w),
+        match display_rotation {
+            &DisplayRotation::Rotate0 | &DisplayRotation::Rotate180 => (w, h),
+            &DisplayRotation::Rotate90 | &DisplayRotation::Rotate270 => (h, w),
         }
     }
 
     /// Set the display rotation
     pub fn set_rotation(&mut self, rot: DisplayRotation) -> Result<(), DI::Error> {
```

Comment:
Can we put common methods like this into a `DisplayMode` (or something) trait that every mode implements? It would make consistent, common behaviour easier to handle I think.

Reply:
Yes, absolutely. This new layer is supposed to enable exactly that. I tried a few ways with traits to just get the common stuff done but some problems I ran into were:
- Switching mode requires releasing all resources and reinitialising it in the new mode (which is not only superfluous and also has a lot of repetition in it and requires a lot of changes if we add something)
- It's not easily possibility to add custom data (e.g. the buffer) to modes requiring it

By putting all common aspects in a custom layer it should be much more easily possible to create the desired DisplayMode trait that provides a common interface allowing to switch between drivers on the fly without having to care about a bucket load of crazy trait dependencies and bounds. Still need to get that 20+ years of C/C++ experience out of my brain... ;)

Reply:
Ok, if you've got plans to clean this up I'm looking forward to seeing them :). Releasing and reinitialising sounds painful so let's not do that haha.

Reply:
Yeah, this stuff really shouldn't be there. Planning to move everything out except for mode specific initialisation, cleanup and stuff.